core/templates/pages/exploration-editor-page/services/exploration-rights-backend-api.service.spec.ts

```diff
@@ -0,0 +1,445 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the exploration rights service
+ * of the exploration editor page.
+ */
+
+import { TestBed, fakeAsync, flushMicrotasks } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { AlertsService } from 'services/alerts.service';
+import { CsrfTokenService } from 'services/csrf-token.service';
+import { ExplorationDataService } from 'pages/exploration-editor-page/services/exploration-data.service';
+import { ExplorationRightsService } from './exploration-rights-backend-api.service';
+
+describe('Exploration rights service', () => {
+  let ers: ExplorationRightsService = null;
+  let als: AlertsService = null;
+  let httpTestingController: HttpTestingController = null;
+  let csrfService = null;
+  let clearWarningsSpy = null;
+  let successHandler = null;
+  let failHandler = null;
+  let serviceData = {
+    rights: {
+      owner_names: ['abc'],
+      editor_names: [],
+      voice_artist_names: [],
+      viewer_names: [],
+      status: 'private',
+      cloned_from: 'e1234',
+      community_owned: true,
+      viewable_if_private: true
+    }
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        {
+          provide: ExplorationDataService,
+          useValue: {
+            explorationId: '12345',
+            data: {
+              version: 1
+            }
+          }
+        }
+      ]
+    });
+    als = TestBed.inject(AlertsService);
+    csrfService = TestBed.inject(CsrfTokenService);
+    ers =
+      TestBed.inject(ExplorationRightsService);
```

Comment:
Can you do the assignment in a single line?

Reply:
Done